flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/types/ResourceProfile.java

```diff
@@ -52,6 +63,15 @@
 	/** How many native memory in mb are needed */
 	private final int nativeMemoryInMB;
 
+	/** A extensible field for user specified resources from {@link ResourceSpec}. */
+	private final Map<String, Double> extendedResources = new TreeMap<>();
```

Comment:
Why not using a `HashMap`?

Reply:
Why is `extendedResources` of type `Map<String, Double` and not `Map<String, Resource>`? I guess it's mostly because we no longer have to merge `Resources`. But what happens if we want to combine multiple `ResourceProfiles` because we want to launch a TM which contains multiple slots (each of them having a single `ResourceProfile` assigned)?

Reply:
HashMap is not ordered, so when comparing two ResourceProfile, the result may differ if the extended resource are added in different order. Using TreeMap can promise the comparision of two ResourceProfile is always same if the extend resources is same.